### DISCUSSION TITLE
Voice navigation irritating when leaving a highway: It shouldn't say the road that you want to leave

### DISCUSSION THREAD

**Discussion by clamyahoo** [osmandapp/OsmAnd#20704] [Ideas]
Hi,
I have a problem with the voice navigation: When I want to leave a road it tells me to do so but when it tells me where to go it first says that I have to go to the road where I am currently on. 
Example: 
![Auswahl_164](https://github.com/user-attachments/assets/c00e4648-e36b-49a4-aacf-b839457cc5a3)
When I come here from southeast an I have to leave the B28, it tells me: "Leave the road on the right hand side - go to B28 ..." and then where I really have to go.
IMHO this shouldn't be that way. I checked if the trunk link contains the name of B28 but it doesn't. Can this part be cut off anyhow?
Kind regards

**Comment by vshcherb** [comment]
Nope it shouldn't , voice navigation says next road from the point of maneuvre. The issue belongs to complex intersections where point of split happens before road changes the name.
